src/api/resources/sync/pull.py

```diff
@@ -20,44 +39,75 @@ def _get_records(model_class, filters):
             revision.revision_num as revision_revision_num,
             revision.deleted as revision_deleted
         FROM
-            revision
-        LEFT JOIN
             model_select
+        INNER JOIN
```

Comment:
Just for my own curiosity, I'd be interested in a walkthrough of why you decided we could no longer do this in one query with left join